matrix/example.lua

```diff
@@ -0,0 +1,24 @@
+return function(s)
+  local rows = {}
+  s:gsub('[%w ]+', function(row_string)
+    local row = {}
+    row_string:gsub('%w+', function(v)
+      table.insert(row, tonumber(v))
+    end)
+    table.insert(rows, row)
+  end)
```

Comment:
I refactored to use `gmatch` (instead of abusing `gsub`) and made some improvements to the names of intermediate variables. Hopefully this helps.